include/circt/Dialect/Emit/EmitOpInterfaces.td

```diff
@@ -0,0 +1,23 @@
+//===- EmitOpInterfaces.td - Operation Interfaces ----------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the Seq operation interfaces.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_EMIT_EMITOPINTERFACES
+#define CIRCT_DIALECT_EMIT_EMITOPINTERFACES
+
+include "mlir/IR/OpBase.td"
+
+def Emittable : OpInterface<"Emittable"> {
```

Comment:
Are you going to use to create methods to the interface? If not, maybe trait is better fit for this?